fs/sync/sync.go

```diff
@@ -482,6 +489,48 @@ func deleteEmptyDirectories(f fs.Fs, entries fs.DirEntries) error {
 	return nil
 }
 
+// This copies the empty directories in the slice passed in and logs
+// any errors copying the directories
+func copyEmptyDirectories(f fs.Fs, entries map[string]fs.DirEntry) error {
+	if len(entries) == 0 {
+		return nil
+	}
+
+	var okCount int
+	for _, entry := range entries {
+		dir, ok := entry.(fs.Directory)
+		if ok {
+			err := f.Mkdir(dir.Remote())
+			if err != nil {
+				fs.Errorf(fs.LogDirName(f, dir.Remote()), "Failed to Mkdir: %v", err)
+				accounting.Stats.Error(err)
+			} else {
+				okCount++
+			}
+		} else {
+			fs.Errorf(f, "Not a directory: %v", entry)
+		}
+	}
+
+	if accounting.Stats.Errored() {
+		fs.Debugf(f, "failed to copy %d directories", accounting.Stats.GetErrors())
+	}
+
+	if okCount > 0 {
+		fs.Debugf(f, "copied %d directories", okCount)
+	}
+	return nil
+}
+
+func parentDirCheck(entries map[string]fs.DirEntry, entry fs.DirEntry) {
+	path := strings.Split(entry.Remote(), "/")
```

Comment:
I think you could replace this

```go
path := strings.Split(entry.Remote(), "/")
path = path[:len(path)-1]
parentDir := strings.Join(path, "/")
```

With this which is what I use elsewhere in rclone which is a bit simpler and will probably run a bit quicker (but who cares!)

```go
parentDir := path.Dir(entry.Remote())
if parentDir == "." {
    parentDir = ""
}
```